### PR TITLE
Feature/react attach file

### PR DESCRIPTION
**Pull request by spykelion**
Send and receive files

### CODE REVIEW

File: React/src/ChatRoom.js [hunk 1]
```diff
@@ -44,23 +50,60 @@ function ChatRoom() {
...
 
+        // upload file here. Default fetch api behavior.
+        if (file) {
+            const data = new FormData();
+            const fileName = file.name;
+            data.append("name", fileName);
+            data.append("file", file);
+
+            try {
+                const response = await fetch(`${WebAPIBaseURL}/api/File`, {
```
Comment (yemelitc) [inline React/src/ChatRoom.js:76]: Why did you not just use the "_fetch()" function?

File: React/src/ChatRoom.js [hunk 2]
```diff
@@ -149,12 +220,17 @@ function ChatRoom() {
...
+                        style={{ display: "none" }}
+                        onChange={(e) => setFile(e.target.files[0])}
+                    />
                     <button onClick={sendMessage} type="submit">
                         Send a message
                     </button>
                 </form>
-                {input.length>0?<Send onClick={sendMessage}/> : <Mic />}
-                
+                {input.length > 0 ? <Send onClick={sendMessage} /> : <Mic />}
```
Comment (yemelitc) [inline React/src/ChatRoom.js:233]: Well, auto-format has done what I commented in PR #8

File: React/src/ChatRoom.js [hunk 3]
```diff
@@ -127,6 +181,24 @@ function ChatRoom() {
                 <div ref={gotoLastMessageRef}></div>
             </div>
 
+            {file && (
+                <div className="shareImgContainer">
```
Comment (yemelitc) [inline React/src/ChatRoom.js:185]: You should probably put this inside a separate function (not necessarily a separate file) then use it here.

File: React/src/ChatRoom.js [hunk 4]
```diff
@@ -106,7 +158,9 @@ function ChatRoom() {
                         <SearchOutlined />
                     </IconButton>
                     <IconButton>
-                        <AttachFile />
+                        <label htmlFor="file">
```
Comment (yemelitc) [inline React/src/ChatRoom.js:161]: Nice!

File: React/src/ChatRoom.js [hunk 5]
```diff
@@ -127,6 +181,24 @@ function ChatRoom() {
...
 
+            {file && (
+                <div className="shareImgContainer">
+                    {/* {file.type == "image/jpeg" || file.type == "image/png"? */}
+                    {/* <img className="shareImg" src={URL.createObjectURL(file)} alt="" /> :  */}
+                    <div>
+                        <AttachFile />
+                        <span>{file.name}</span>
+                    </div>
+                    {/* } */}
```
Comment (yemelitc) [inline React/src/ChatRoom.js:192]: What is this line doing?

File: React/src/Message.js [hunk 6]
```diff
@@ -31,19 +32,25 @@ function Message({ messages, message, roomInfo, setLinkedId }) {
...
             ) : null}
 
             {message.file ? (
-                <div className="chat__image">
-                    <div className="chat__imageLink">
-                        <img src={getFileURL(message.file.name)} alt="" />
-                    </div>
-                </div>
+                <>
+                    {getFileURL(message.file.name).match(extensions) ? (
```
Comment (yemelitc) [inline React/src/Message.js:42]: Not sure what is going on here. It should have been a separate commit. I will test later.

File: React/src/ChatRoom.js (hunk 1 above)
Comment (spykelion) [reply React/src/ChatRoom.js:76]: The content type restricted files.
https://github.com/estenligne/ChatOnline/blob/main/React/src/global.js#L20

Comment (yemelitc) [reply React/src/ChatRoom.js:76]: Oh I see, you are right. That means that _fetch function needs to be fixed. I will do so in a separate commit on this branch.

File: React/src/ChatRoom.js (hunk 3 above)
Comment (yemelitc) [reply React/src/ChatRoom.js:185]: I see this is small enough to not need to put in a separate function. I will remove the comments, I suppose they are not necessary.

File: React/src/Message.js (hunk 6 above)
Comment (yemelitc) [reply React/src/Message.js:42]: I have tested and now I understand, a good idea you implemented there!

File: React/src/ChatRoom.js (hunk 5 above)
Comment (yemelitc) [reply React/src/ChatRoom.js:192]: I have removed the line alongside the other comment-only lines.

**Review by yemelitc — APPROVED** [review]

**Pull request merged**